### ISSUE TITLE
Can't split VCF file containing HLA

### ISSUE THREAD

**Issue by Akazhiel** [pcingola/SnpEff#265]
As the title says, I'm getting the following error when trying to split a VCF file which has contigs with names like HLA-A*01:03

```
Exception in thread "main" java.lang.RuntimeException: Error opening file 'combined_calls.HLA-A*01:03.vcf'
	at org.snpsift.SnpSiftCmdSplit.newFile(SnpSiftCmdSplit.java:106)
	at org.snpsift.SnpSiftCmdSplit.split(SnpSiftCmdSplit.java:197)
	at org.snpsift.SnpSiftCmdSplit.run(SnpSiftCmdSplit.java:150)
	at org.snpsift.SnpSift.run(SnpSift.java:588)
	at org.snpsift.SnpSift.main(SnpSift.java:76)
```

**Comment by pcingola** [comment]
Can you send me the command line you are using, and if possible a couple of line from yor VCF file?

**Comment by Akazhiel** [comment]
On your reply I just realised you have two different repositories for SnpEff and SnpSift, this actually happened when running SnpSift. The command was the following `java -jar SnpSift.jar split combined_calls.vcf` .

And the lines giving problem are ones like the following:

```
HLA-DRB1*14:54:01	13862	.	C	T	.	PASS	AC=1;ADP=33;AF=0.500;AN=2;HET=1;HOM=0;NC=0;WT=0;set=varscan	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	./.	0/1:28:2:1:1:33:6.06%:0:9.8E-1:32:31:6:25:33
HLA-DRB1*14:54:01	13863	.	C	T	.	PASS	AC=1;ADP=31;AF=0.500;AN=2;HET=1;HOM=0;NC=0;WT=0;set=varscan	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	./.	0/1:20:1:0:1:31:3.23%:0:9.8E-1:34:30:7:23:31
HLA-DRB1*14:54:01	13866	.	CTG	C	.	PASS	AC=1;ADP=30;AF=0.500;AN=2;HET=1;HOM=0;NC=0;WT=0;set=varscan	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	./.	0/1:27:1:0:1:30:3.33%:0:9.8E-1:33:28:6:22:30
HLA-DRB1*14:54:01	13874	.	C	T	.	PASS	AC=1;ADP=25;AF=0.500;AN=2;HET=1;HOM=0;NC=0;WT=0;set=varscan	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	./.	0/1:19:2:0:2:25:8%:0:9.8E-1:31:23:3:20:25
HLA-DRB1*14:54:01	13879	.	C	T	.	PASS	AC=1;ADP=25;AF=0.500;AN=2;HET=1;HOM=0;NC=0;WT=0;set=varscan	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	./.	0/1:19:1:0:1:25:4%:0:9.8E-1:26:24:0:24:25
HLA-DRB1*14:54:01	13883	.	C	T	.	PASS	AC=1;ADP=21;AF=0.500;AN=2;HET=1;HOM=0;NC=0;WT=0;set=varscan	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	./.	0/1:19:2:0:2:21:9.52%:0:9.8E-1:27:19:0:19:21
HLA-DRB1*14:54:01	13884	.	C	T	.	PASS	AC=1;ADP=19;AF=0.500;AN=2;HET=1;HOM=0;NC=0;WT=0;set=varscan	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	./.	0/1:18:1:0:1:19:5.26%:0:9.8E-1:26:18:0:18:19
HLA-DRB1*14:54:01	13896	.	A	G	.	PASS	AC=1;ADP=9;AF=0.500;AN=2;HET=1;HOM=0;NC=0;WT=0;set=varscan	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	./.	0/1:25:1:0:1:9:11.11%:0:9.8E-1:26:8:0:8:9
HLA-DRB1*14:54:01	13915	.	G	T	.	PASS	AC=1;ADP=6;AF=0.500;AN=2;HET=1;HOM=0;NC=0;WT=0;set=varscan	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	./.	0/1:19:1:0:1:6:16.67%:0:9.8E-1:26:5:0:5:6
HLA-DRB1*14:54:01	13921	.	C	T	.	PASS	AC=1;ADP=6;AF=0.500;AN=2;HET=1;HOM=0;NC=0;WT=0;set=varscan	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	./.	0/1:16:1:0:1:6:16.67%:0:9.8E-1:26:5:0:5:6
```

**Comment by pcingola** [comment]
Works OK for me:

```
$ java -jar SnpSift.jar split -v combined_calls.vcf
00:00:00	SnpSift version 5.0b (build 2020-10-16 14:33), by Pablo Cingolani
00:00:00	Command: 'split'
00:00:00	Splitting file 'combined_calls.vcf'

00:00:00	Creating new file 'combined_calls.HLA-DRB1*14:54:01.vcf'


$ ls combined_calls.*vcf
combined_calls.HLA-DRB1*14:54:01.vcf	combined_calls.vcf
```

Which platform /operating system are you using?

**Comment by Akazhiel** [comment]
I am using a Linux system running Ubuntu. Just attempted to run it again, with the exact same command, and got the same error.

**Comment by pcingola** [comment]
Fixed in v5.0c
replaces non alphanumeric chars by '_' for creating the file names:
```
$ java -jar SnpSift.jar split -v combined_calls.vcf
00:00:00	SnpSift version 5.0c (build 2020-11-24 09:59), by Pablo Cingolani
00:00:00	Command: 'split'
00:00:00	Splitting file 'combined_calls.vcf'

00:00:00	Creating new file 'combined_calls.HLA-DRB1_14_54_01.vcf'

$ ls combined_calls.*
combined_calls.HLA-DRB1_14_54_01.vcf	combined_calls.vcf
```
